src/docs/commands_keybindings.md

```diff
@@ -1,122 +1,142 @@
 
-**java-frontend-module.ts**
+**helloworld-frontend-module.ts**
 ```typescript
 export default new ContainerModule(bind => {
-    bind(CommandContribution).to(JavaCommandContribution).inSingletonScope();
-    ...
+   // add your contribution bindings here
+   bind(CommandContribution).to(HelloworldCommandContribution);
+...
 });
 ```
 
+The `CommandRegistry` used to register our contributed command above, also provides an API to interact with commands. As an example, you can programmatically execute commands, you can browse through all registered commands or you can access a list of recently executed commands. Please refer to the [TypeDoc for the CommandRegistry](https://eclipse-theia.github.io/theia/docs/next/classes/core.commandregistry-1.html) for more details. To use the `CommandRegistry` outside of a contribution, you can access it via dependency injection.
 
-The class that is in charge of registering and executing the commands is the command registry. It is also possible to get the list of commands with the `get commandIds()` api.
+In the following sections, we describe how to bind commands to menu items and keybindings.
 
-## Adding keybindings to the keybinding registry
+## Contributing Menu Items
 
-Commands don't need keybindings by default, as they can be called by a lot of other means (programmatically, user clicks etc.). However, keybindings with specific contexts can be added to a command in a similar way to contributing commands.
+Theia allows you to contribute menu items that will be displayed in specific menus within your Theia application. Menu items are bound to commands and therefore allow the user to trigger actions (please see the section about commands above).
 
-To contribute a keybinding, simply inject an implementation of `KeybindingContribution`.
+All the following code examples are from the [Theia extension generator](https://github.com/eclipse-theia/generator-theia-extension). You can get the same code set-up by installing the generator, selecting the “Hello World” example (see [here](https://github.com/eclipse-theia/generator-theia-extension)) and choosing “helloworld” as the name.
 
-**editor-keybinding.ts**
+All menu items of a Theia application are managed in the `MenuModelRegistry`. To contribute menu items to the registry, modules must implement the ´MenuContribution´ interface (see code example below). 
+
+The registration of the command can be done in the function `registerMenus`, which will be called by the Theia framework. The function provides the `MenuModelRegistry` as a parameter. On this registry we can call `registerMenuAction`. It expects a `MenuPath` and a `MenuAction`. The `MenuPath` specifies the menu (and submenu) to place the menu item into. Please see [here for the paths of some common menus](https://github.com/eclipse-theia/theia/blob/f776392b6b616af48d222fbeb035e08c2c1f2865/packages/core/src/browser/common-frontend-contribution.ts#L59-L85).
```

Comment:
> Please see [here for the paths of some common menus](https://github.com/eclipse-theia/theia/blob/f776392b6b616af48d222fbeb035e08c2c1f2865/packages/core/src/browser/common-frontend-contribution.ts#L59-L85).

We can also use refer to the TypeDoc directly instead:
- https://eclipse-theia.github.io/theia/docs/next/modules/core.commonmenus-1.html